### DISCUSSION TITLE
Amazing resource, one piece of feedback for install NixOS with flakes

### DISCUSSION THREAD

**Discussion by tacomilkshake** [ryan4yin/nixos-and-flakes-book#46] [Ideas]
I love this book. It's without a doubt the best resource that now exists for learning Nix.

One piece of feedback for install NixOS with flakes... when I install a new NixOS system with flakes, I never end-up using anything but flakes. Upon booting into the NixOS installer via a USB drive, I copy over the git repo for the flake/nix config, and run:

sudo nixos-install --no-root-passwd --flake ./nix-config/#hostname --impure

This allows me to install NixOS without ever having to deal with a configuration.nix file. You will need to make sure the hardware config is in your Git repo.

**Comment by ryan4yin** [comment]
Thanks for the feedback, I do know that such an installation command exists, but considering that this method adds some complexity and may not be friendly to beginners, I did not describe it in this article.

**Reply by tacomilkshake** [reply]
That makes sense. Thanks again for providing this exceptional resource. I plan to share it a lot!
